### PR TITLE
CI based on Github workflows

### PR DESCRIPTION
**Pull request by msoeken**

### CODE REVIEW

**Comment by msoeken** [comment]
@hriener @boschmitt This PR can change CI from Travis/AppVeyor to Github Actions. Do you agree with the current configurations (Linux GCC 7, 8, 9, Clang 8, 9), MacOS (GCC 8, 9, Clang 9), and Windows (VS 2019) and to remove Travis and AppVeyor integration?

**Comment by msoeken** [comment]
Build and test times seem to be much faster.

**Comment by hriener** [comment]
Can we also integrate coverage analysis?

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+jobs:
+  build-gcc8:
+    name: Build with GNU GCC 8
+    runs-on: macOS-latest
+
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: true
+    - name: Build kitty
```
Comment (marcelwa) [inline .github/workflows/macos.yml]: Did you mean "Build mockturtle" here? (same below)

Comment (msoeken) [reply .github/workflows/macos.yml]: Thanks, great catch!

**Comment by msoeken** [comment]
> Can we also integrate coverage analysis?

I don't think we can have a nice visual representation of coverage results compared to other tools such as coveralls. But for checking whether the build is successful or not, Github Actions seem to be just as good as Travis and AppVeyor, at least for our current configuration. For coverage we should have a separate PR and adapt from what has been done for Lorna.

**Comment by hriener** [comment]
> > Can we also integrate coverage analysis?

> Github Actions seem to be just as good as Travis and AppVeyor, at least for our current configuration.

If it is "as good" and not "better", why do we care? (I'm not against, only wondering.)

> For coverage we should have a separate PR and adapt from what has been done for Lorna.

My main question is if it would work to integrate it as before?

**Comment by msoeken** [comment]
> > > Can we also integrate coverage analysis?
> 
> > Github Actions seem to be just as good as Travis and AppVeyor, at least for our current configuration.
> 
> If it is "as good" and not "better", why do we care? (I'm not against, only wondering.)
> 

Actually, it is much faster in Github. Right now it takes about an hour with Travis, but only max. 10 minutes in Github.

> > For coverage we should have a separate PR and adapt from what has been done for Lorna.
> 
> My main question is if it would work to integrate it as before?

Does the lorina coverage integration depend on Travis or AppVeyor?

**Comment by msoeken** [comment]
@hriener I took a look at the coveralls integration in lorina. We can do the same thing with Github Actions.

**Comment by hriener** [comment]
Reducing compile-time is an important plus point.

Coverage analysis with coveralls in `lorina` uses only three modifications in Travis:
* Installation of `cpp-coveralls` before compilation
* Some additional flags for compilation
* Execution of `coveralls` after compilation

**Comment by msoeken** [comment]
> Reducing compile-time is an important plus point.
> 
> Coverage analysis with coveralls in `lorina` uses only three modifications in Travis:
> 
> * Installation of `cpp-coveralls` before compilation
> * Some additional flags for compilation
> * Execution of `coveralls` after compilation

We can all do that in Github Actions.

**Pull request merged**